apps/src/templates/studioHomepages/Courses.jsx

```diff
@@ -41,6 +41,36 @@ class Courses extends Component {
       .show();
   }
 
+  getHeroStrings() {
+    const {isTeacher, isSignedOut, studentsCount} = this.props;
+
+    // Default to "Learn" view strings
+    let heroStrings = {
+      headingText: i18n.coursesLearnHeroHeading(),
+      subHeadingText: i18n.coursesLearnHeroSubHeading({studentsCount}),
+      description: i18n.coursesLearnHeroDescription(),
+      buttonText: i18n.coursesLearnHeroButton()
+    };
+
+    // Apply overrides if this is the "Teach" view
+    if (isTeacher) {
+      heroStrings = {
+        headingText: i18n.coursesTeachHeroHeading(),
+        subHeadingText: i18n.coursesTeachHeroSubHeading(),
+        description: i18n.coursesTeachHeroDescription(),
+        buttonText: i18n.coursesTeachHeroButton()
+      };
+    }
+
+    // We show a short version of the banner when you're signed in,
+    // so don't include the description string.
+    if (!isSignedOut) {
+      delete heroStrings.description;
```

Comment:
I wonder if it would be cleaner to only add the description if we are signed-out?

Reply:
Yeah, it might be.  This was the only thing that came to mind that doesn't nest an `isTeacher` and `isSignedOut` check somehow, but what your suggestion is almost as DRY and probably easier to read.  I'll change it up.